src/gl/implementation/gl-objects-buffers.adb

```diff
@@ -46,6 +55,25 @@ package body GL.Objects.Buffers is
       end if;
    end Bind;
 
+   procedure Bind_Transform_Feedback (Object : Transform_Buffer'Class) is
+      Target_Kind : constant GL.Low_Level.Enums.Buffer_Kind :=
+                        GL.Low_Level.Enums.Transform_Feedback;
+      Cursor : constant Transform_Buffer_Maps.Cursor
+        := Current_Transform_Buffers.Find (Target_Kind);
+   begin
+      if Cursor = Transform_Buffer_Maps.No_Element or else
+        Transform_Buffer_Maps.Element (Cursor).Reference.GL_Id /= Object.Reference.GL_Id
```

Comment:
Why should the procedure check this? Your code basically checks whether the same ID is not already bound, however if it is, the call to `Bind_Transform_Feedback` would simply do nothing, so there is no reason to guard against it. Using `Indefinite_Hashed_Maps` is a significant overhead that is not justified by what you're doing here.